### PR TITLE
tools/rados: add --pgid in help

### PR DESCRIPTION
**Pull request by vumrao**
tools/rados: add --pgid in help
This commit 825c7b9 brought the support for --pgid in rados ls but was not in help.
Fixes: https://tracker.ceph.com/issues/41876
Signed-off-by: Vikhyat Umrao <vumrao@redhat.com>

### CODE REVIEW

File: doc/man/8/rados.rst [hunk 1]
```diff
@@ -199,6 +203,10 @@ To get a list object in pool foo sent to stdout::
 
        rados -p foo ls -
 
+To get a list of objects in PG 0.6::
+
+       rados ls --pgid 0.6
```
Comment (tchaikov) [inline doc/man/8/rados.rst]: could you move `--pgid 0.6` before `ls` ? as this option is not specific to `ls` command.

File: doc/man/8/rados.rst [hunk 2]
```diff
@@ -26,6 +26,10 @@ Options
 
    Interact with the given pool. Required by most commands.
 
+.. option:: --pgid
+
+   Interact with the given PG id. Required by rados ls command.
```
Comment (tchaikov) [inline doc/man/8/rados.rst]: it's not "required", also it applies to all commands which uses an io_ctx in the rados cli tool. we probably could put this way

> As an alternative to `--pool`, `--pgid` also allows user to specify the PG id to specify the pool to which the command will be directed. With this option, certain commands like `ls` allows user to limit the scope of the command. See

and we need to revise the help of `ls` command

> :command:`ls` *outfile*
>     List objects in given pool and write to outfile. if `--pgid` is specified, `ls` will only list the objects in the given PG.

File: src/tools/rados/rados.cc [hunk 3]
```diff
@@ -176,6 +176,8 @@ void usage(ostream& out)
 "        select given pool by name\n"
 "   --target-pool=pool\n"
 "        select target pool by name\n"
+"   --pgid pg id\n"
+"        select given pg id\n"
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: we are using both "PG" and "pg" interchangeably in the help message. could you take this opportunity to unify them? i'd suggest use "PG" in the user visible document and messages. as it's an abbreviation  of placement group. what do you think?

**Comment by tchaikov** [comment]
retest this please.

**Comment by vumrao** [comment]
Thanks, @tchaikov for feedback. I have made the suggested changes.

File: doc/man/8/rados.rst [hunk 4]
```diff
@@ -104,7 +110,7 @@ Pool specific commands
   List the watchers of object name.
 
 :command:`ls` *outfile*
-  List objects in given pool and write to outfile.
+  List objects in the given pool and write to outfile. Instead of --pool if --pgid will be specified, ls will only list the objects in the given PG.
```
Comment (tchaikov) [inline doc/man/8/rados.rst]: could you quote "--pool", "--pgid" and "ls" with backquotes?

File: doc/man/8/rados.rst [hunk 5]
```diff
@@ -26,6 +26,12 @@ Options
 
    Interact with the given pool. Required by most commands.
 
+.. option:: --pgid
+
+   As an alternative to --pool, --pgid also allows user to specify the
+   PG id to which the command will be directed. With this option, certain
+   commands like ls allows user to limit the scope of the command to the given PG.
```
Comment (tchaikov) [inline doc/man/8/rados.rst]: could you quote "--pool", "--pgid" and "ls" with backquotes?

Comment (vumrao) [reply doc/man/8/rados.rst]: thanks Kefu. done.

File: doc/man/8/rados.rst (hunk 4 above)
Comment (vumrao) [reply doc/man/8/rados.rst]: added.

File: doc/man/8/rados.rst [hunk 6]
```diff
@@ -26,6 +26,12 @@ Options
 
    Interact with the given pool. Required by most commands.
 
+.. option:: --pgid
+
+   As an alternative to "--pool", "--pgid" also allows user to specify the
+   PG id to which the command will be directed. With this option, certain
+   commands like "ls" allows user to limit the scope of the command to the given PG.
+
```
Comment (tchaikov) [inline doc/man/8/rados.rst]:
```diff
--- a/doc/man/8/rados.rst
+++ b/doc/man/8/rados.rst
@@ -28,9 +28,9 @@ Options

 .. option:: --pgid

-   As an alternative to "--pool", "--pgid" also allows user to specify the
+   As an alternative to ``--pool``, ``--pgid`` also allows user to specify the
    PG id to which the command will be directed. With this option, certain
-   commands like "ls" allows user to limit the scope of the command to the given PG.
+   commands like ``ls`` allows user to limit the scope of the command to the given PG.

 .. option:: -s snap, --snap snap
```

Comment (vumrao) [reply doc/man/8/rados.rst]: sorry Kefu, misunderstood the quotes, fixed it. I have also verified with hint given in the comment - make manpages and it looks good.

File: doc/man/8/rados.rst [hunk 7]
```diff
@@ -26,6 +26,12 @@ Options
 
    Interact with the given pool. Required by most commands.
 
+.. option:: --pgid
+
+   As an alternative to ``--pool``, ``--pgid`` also allows user to specify the
```
Comment (neha-ojha) [inline doc/man/8/rados.rst]:
```suggestion
   As an alternative to ``--pool``, ``--pgid`` also allows users to specify the
```

File: doc/man/8/rados.rst [hunk 8]
```diff
@@ -26,6 +26,12 @@ Options
...
+   PG id to which the command will be directed. With this option, certain
+   commands like ``ls`` allows user to limit the scope of the command to the given PG.
```
Comment (neha-ojha) [inline doc/man/8/rados.rst]:
```suggestion
   commands like ``ls`` allows users to limit the scope of the command to the given PG.
```

File: doc/man/8/rados.rst (hunk 7 above)
Comment (vumrao) [reply doc/man/8/rados.rst]: Done.

File: doc/man/8/rados.rst (hunk 8 above)
Comment (vumrao) [reply doc/man/8/rados.rst]: Done.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-21_17:17:30-rados-wip-kefu-testing-2019-09-20-1944-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-21_17:17:30-rados-wip-kefu-testing-2019-09-20-1944-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-22_03:04:52-rados-wip-kefu-testing-2019-09-20-1945-distro-basic-mira/

failures tracked by

- https://tracker.ceph.com/issues/41744
- https://tracker.ceph.com/issues/41946

**Pull request merged**